### PR TITLE
[UXP-20366] Addition of Overlay-0.37.0

### PR DESCRIPTION
**Pull request by deepthi-majji**
Adding wrapper for overlay. 

## Types of changes

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

-   [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
-   [ ] My code follows the code style of this project.
-   [ ] My change requires a change to the documentation.
-   [ ] I have updated the documentation accordingly.
-   [ ] I have read the **CONTRIBUTING** document.
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/overlay/package.json [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    "module": "src/index.js",
+    "type": "module",
+    "exports": {
+        ".": "./src/index.js",
+        "./package.json": "./package.json",
+        "./src/Overlay.js": "./src/Overlay.js",
+        "./src/overlay.css.js": "./src/overlay.css.js",
+        "./sp-overlay.js": "./sp-overlay.js"
+    },
+    "dependencies": {
```
Comment (inasirhussain) [inline packages/overlay/package.json:25]: We should specify transitive dependency as well (those which we have written wrappers for)- https://github.com/adobe/spectrum-web-components/blob/v0.37.0/packages/overlay/package.json#L131

File: packages/overlay/package.json [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    "repository": {
+        "type": "git",
+        "url": "https://git.corp.adobe.com/torq/swc-uxp-wrappers.git",
+        "directory": "packages/overlay"
+    },
+    "author": "Adobe Inc",
+    "main": "src/index.js",
+    "module": "src/index.js",
+    "type": "module",
+    "exports": {
```
Comment (inasirhussain) [inline packages/overlay/package.json:18]: Why are we not exporting OverlayTrigger etc.? 
https://github.com/adobe/spectrum-web-components/blob/v0.37.0/packages/overlay/package.json#L22

**Comment by inasirhussain** [comment]
@deepthi-majji - Please add aliasing changes in utils/alias.js

File: packages/overlay/package.json (hunk 1 above)
Comment (deepthi-majji) [reply packages/overlay/package.json:28]: added it.

File: packages/overlay/package.json (hunk 2 above)
Comment (deepthi-majji) [reply packages/overlay/package.json:18]: added it

File: projects/swc-starter-webpack/src/extended_swc_samples/my-overlay.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+/*
+Copyright 2023 Adobe. All rights reserved.
```
Comment (inasirhussain) [inline projects/swc-starter-webpack/src/extended_swc_samples/my-overlay.js:2]:
```suggestion
Copyright 2024 Adobe. All rights reserved.
```

**Comment by inasirhussain** [comment]
@deepthi-majji - Please run `yarn format:fix` to fix formatting error in the PR.

File: packages/overlay/package.json [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        "./src/loader.js": "./src/loader.js",
+        "./src/overlay-events.js":  "./src/overlay-events.js",
+        "./src/overlay-timer.js":  "./src/overlay-timer.js",
+        "./src/overlay-trigger.css.js": "./src/overlay-trigger.css.js",
+        "./src/overlay-types.js": "./src/overlay-types.js",
+        "./src/overlay.css.js": "./src/overlay.css.js",
+        "./src/topLayerOverTransforms.js": "./src/topLayerOverTransforms.js",
+        "./active-overlay.js":  "./active-overlay.js",
+        "./overlay-trigger.js": "./overlay-trigger.js",
+        "./sync/overlay-trigger.js":  "./overlay-trigger-sync.js",
```
Comment (inasirhussain) [inline packages/overlay/package.json:41]: we dont have any folder sync and file overlay-trigger.js?

Comment (deepthi-majji) [reply packages/overlay/package.json:41]: We don't have it but it is there in swc folder structure so I just re-exported to the file I have created inside our package.

**Pull request merged**